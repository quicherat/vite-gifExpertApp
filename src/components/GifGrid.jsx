import { useEffect, useState } from "react";
import { getGifs } from "../helpers/getGifs.js";
import { GifItem } from "./GifItem.jsx";



const GifGrid = ({category}) => { 

  const [images, setImages] = useState([]);

  const getImages = async() => {
    const newImages = await getGifs( category );
    setImages(newImages);
  }

  useEffect( () => {
    getImages(); 
  }, []);

  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">
        {images.map( (image) => 
          (<GifItem 
            key={image.id}
            { ...image }
            // tambiÃ©n con image={image} van todas las propiedades
          />)
        )}        
      </div>
      
    </>
  )
}

export default GifGrid
