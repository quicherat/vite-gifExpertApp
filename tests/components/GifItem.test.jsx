import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"


describe('Pruebas para GifItem', () => {
  
    const title = 'titulo prueba';
    const url = 'https://giphy.com/saitama.jpg';

    test('Prueba del snapshot', () => {
        

        const { container } = render(<GifItem title={title} url={url} />);

        expect (container). toMatchSnapshot();
    })

    test('Debe mostrar la imagen con el url y el alt indicado ', () => {
      
        render(<GifItem title={title} url={url} />);
        // screen.debug();
        // expect( screen.getByRole('img').src).toBe(url);
        // expect( screen.getByRole('img').alt).toBe(title);

        const {src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );

    })

    test('Debe mostrar el tÃ­tulo en el componente ', () => {
      
        render(<GifItem title={title} url={url} />);
        expect( screen.getByText( title )).toBeTruthy();
    })
    
    
    


})
