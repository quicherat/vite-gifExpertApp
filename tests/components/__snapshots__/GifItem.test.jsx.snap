// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para GifItem Prueba del snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="titulo prueba"
      src="https://giphy.com/saitama.jpg"
    />
    <p>
      titulo prueba
    </p>
  </div>
</div>
`;
